swfiles/@spinw/fitpow.m

```diff
@@ -0,0 +1,668 @@
+function fitsp = fitpow(obj, data, varargin)
+% fits experimental powder spin wave data
+%
+% ### Syntax
+%
+% `fitsp = fitpow(obj, data, Name, Value)`
+%
+% ### Description
+% `fitsp = fitpow(obj, data, Name, Value)` takes a set of energy cuts of powder
+% spin wave spectrum and fits this.
+%
+% ### Input Arguments
+%
+% `obj`
+% : [spinw] object.
+%
+% `data`
+% : input data cuts which may be an array of any of:
+%   * a Horace 1D sqw or d1d object
+%   * a struct with fields 'x', 'y', 'e', 'qmin', 'qmax'
+%   * a struct with fields 'IX', 'qmin', 'qmax'
+%   * a cell array {x y e qmin qmax}
+%   * a cell array {IX_dataset_1d qmin qmax}
+%   where x is a vector of energy transfer,
+%         y is a vector of intensity,
+%         e is a vector of uncertainties (errorbar)
+%         qmin is a scalar denoting the lower Q value of the cut
+%         qmax is a scalar denoting the upper Q value of the cut
+%         IX / IX_dataset_1d is a Horace IX_dataset_1D object
+%   Instead of the pair of scalars qmin, qmax, you specify a
+%   2-vector: [qmin qmax] under the key 'qrange', e.g.
+%   * a struct with fields 'x', 'y', 'e', 'qrange'
+%   * a cell array {x y e [qmin qmax]}
+%
+% ### Name-Value Pair Arguments
+%
+% `'func'`
+% : Function to change the Hamiltonian in `obj`, it needs to have the
+%   following header:
+%   ```
+%   obj = @func(obj, x)
+%   ```
+%
+% `'xmin'`
+% : Lower limit of the optimisation parameters, optional.
+%
+% `'xmax'`
+% : Upper limit of the optimisation parameters, optional.
+%
+% `'x0'`
+% : Starting value of the optimisation parameters. If empty
+%  or undefined, random values are used within the given limits.
+%
+% `'optimizer'`
+% : String that determines the type of optimizer to use, possible values:
+%   * `'lm3'`       (Default) Levenberg-Marquardt as implemented in Horace, see [ndbase.lm3],
+%   * `'pso'`       Particle-swarm optimizer, see [ndbase.pso],
+%   * `'simplex'`   Matlab built-in simplex optimizer, see [fminsearch](www.mathworks.ch/help/matlab/ref/fminsearch.html).
+%
+% `'nRun'`
+% : Number of consecutive fitting runs, each result is saved in the
+%   output `fitsp.x` and `fitsp.R` arrays. If the Hamiltonian given by the
+%   random `x` parameters is incompatible with the ground state,
+%   those `x` values will be omitted and new random `x` values will be
+%   generated instead. Default value is 1.
+%
+% `'nMax'`
+% : Maximum number of runs, including the ones that produce error
+%   (due to incompatible ground state). Default value is 1000.
+%
+% `'nQ'`
+% : Number of |Q| points to use for each cut to simulate cut thickness
```

Comment:
What would happen is the cut thickness is << d|Q| resolution (I'm aware that might not be the same for all Q-vectors of equal |Q| - e.g. depends on scattering angle)?

Reply:
Maybe for each Q point in average, you also simulate a number of points inside the resolution ellipsoid? Would that work?

Reply:
Or do you do the resolution convolution on the simulated data before the powder averaging?